docs/ConcurrencyRoadmap.md

```diff
@@ -0,0 +1,265 @@
+
+## Actor Isolation and The Second Phase
+
+The goal is for Swift to prevent data races on mutable state by default. The system through which this is achieved is called **actor isolation**, both because actors are centrally important to how the system works and because the system is centrally concerned with preventing actor-protected state from being accessed outside the actor. However, actor isolation restricts code even when actors aren’t directly involved, when it is necessary for the correctness of the system under concurrency.
+
+We intend to introduce the features described in this roadmap in two phases: first by introducing the ability to create async functions and actors; and second, by enforcing full actor isolation.
+
+The basic idea of actor isolation is similar to the idea of [exclusive access to memory](https://github.com/apple/swift-evolution/blob/main/proposals/0176-enforce-exclusive-access-to-memory.md), and builds upon it. Swift’s concurrency design aims to provide an easy-to-use and composable approach to safe concurrency by starting from the natural isolation of actors and then using ownership as a complementary tool.
+
+The problem of actor isolation reduces to the problem of ensuring that all ordinary mutable memory is accessed only by a particular actor or task.  This in turn reduces to an analysis of how the memory is accessed and who can access it in the first place. We can categorize memory into a few groups:
+
+* Properties of an actor will be protected by the actor.
+* Immutable memory (such as a `let` constant), local memory (such as a local variable that’s never captured), and value component memory (such as a properties of a struct, or an enum case), are already protected from data races.
+* Unsafe memory (such as an arbitrary allocation referenced by an `UnsafeMutablePointer`) is associated with an unsafe abstraction.  It’s actively undesirable to try to force these abstractions to be used safely, because these abstractions are meant to be usable to bypass safe language rules when necessary. Instead, we have to trust the programmer to use these correctly.
+* Global memory (such as a global or static variable) can in principle be accessed by any code anywhere, so is subject to data races.
+* Class component memory can also be accessed from any code that hold a reference to the class. This means that while the _reference_ to the class may be protected by an actor, passing that reference between actors exposes its properties to data races. Note that this also references to classes held within value types when these are passsed between actors.
```

Comment:
Typo "passsed", and I think that last sentence should start with "This also includes references to classes held within value types"

Reply:
```suggestion
* Class component memory can also be accessed from any code that hold a reference to the class. This means that while the _reference_ to the class may be protected by an actor, passing that reference between actors exposes its properties to data races. This also includes references to classes held within value types, when these are passed between actors.
```